### PR TITLE
Upgrade sulu skeleton with symfony recipe defaults

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Upgrade sulu skeleton with symfony recipe defaults.

#### Why?

Matching the defaults of symfony recipes. We did forget that in the last minor releases.

### CODE REVIEW

File: config/packages/framework.yaml [hunk 1]
```diff
@@ -1,17 +1,25 @@
+# see https://symfony.com/doc/current/reference/configuration/framework.html
 framework:
     secret: '%env(APP_SECRET)%'
     #csrf_protection: true
-    #http_method_override: true
+    #http_method_override: false
```
Comment (alexander-schranz) [inline config/packages/framework.yaml:5]: this is different to symfony but we actually enable http_method_override in our `public/index.php`

File: config/packages/prod/doctrine.yaml [hunk 2]
```diff
@@ -1,9 +1,6 @@
 doctrine:
     orm:
         auto_generate_proxy_classes: false
-        metadata_cache_driver:
-            type: pool
-            pool: doctrine.system_cache_pool
```
Comment (alexander-schranz) [inline config/packages/prod/doctrine.yaml:6]: Related to: https://github.com/symfony/recipes/pull/996

File: config/packages/prod/monolog.yaml [hunk 3]
```diff
@@ -8,8 +8,9 @@ monolog:
             buffer_size: 50 # How many messages should be saved? Prevent memory leaks
         nested:
             type: stream
-            path: "%kernel.logs_dir%/%kernel.environment%.log"
+            path: "%kernel.logs_dir%/%kernel.environment%.log" # php://stderr
```
Comment (alexander-schranz) [inline config/packages/prod/monolog.yaml:11]: This is different to the new symfony default as I thought the symfony default doesn't make sense for us, as most of our users have more traditional setup then a cloud setup:  See https://github.com/symfony/recipes/pull/894

File: .env.test [hunk 4]
```diff
@@ -1,4 +1,4 @@
 # define your env variables for the test env here
 KERNEL_CLASS='App\Kernel'
 APP_SECRET='s$cretf0rt3st'
-DATABASE_URL=mysql://db_user:db_password@127.0.0.1:3306/db_name_test?serverVersion=5.7
```
Comment (alexander-schranz) [inline .env.test:4]: this is now set in the `config/packages/test/doctrine.yaml` via a `_test` prefix

File: bin/phpunit [hunk 5]
```diff
@@ -1,13 +1,19 @@
 #!/usr/bin/env php
 <?php
 
-if (!file_exists(dirname(__DIR__).'/vendor/symfony/phpunit-bridge/bin/simple-phpunit.php')) {
-    echo "Unable to find the `simple-phpunit.php` script in `vendor/symfony/phpunit-bridge/bin/`.\n";
-    exit(1);
+if (!ini_get('date.timezone')) {
```
Comment (alexander-schranz) [inline bin/phpunit:4]: Symfony now recommend now to use phpunit: https://github.com/symfony/recipes/pull/906

This has the advantage of not longer have hacky solutions for phpunit autoloading for phpstan and rector.

File: bin/phpunit [hunk 6]
```diff
@@ -1,13 +1,19 @@
...
+    ini_set('date.timezone', 'UTC');
 }
 
-if (false === getenv('SYMFONY_PHPUNIT_DIR')) {
-    putenv('SYMFONY_PHPUNIT_DIR='.__DIR__.'/.phpunit');
-}
+if (is_file(dirname(__DIR__).'/vendor/phpunit/phpunit/phpunit')) {
```
Comment (alexander-schranz) [inline bin/phpunit:8]: Symfony now recommend to use phpunit directly without it being installed over the bridge: https://github.com/symfony/recipes/pull/906

This has the advantage of not longer have hacky solutions for phpunit autoloading for phpstan and rector.

File: config/packages/prod/deprecations.yaml [hunk 7]
```diff
@@ -5,4 +5,4 @@
 #        deprecation:
 #            type: stream
 #            channels: [deprecation]
-#            path: "%kernel.logs_dir%/%kernel.environment%.deprecations.log"
+#            path: "%kernel.logs_dir%/%kernel.environment%.deprecations.log" # php://stderr
```
Comment (alexander-schranz) [inline config/packages/prod/deprecations.yaml:8]: see https://github.com/sulu/skeleton/pull/150/files#r766826100

File: .env.test (hunk 4 above)
Comment (wachterjohannes) [reply .env.test:4]: strange 🙈  but OK for me

File: bin/phpunit (hunk 6 above)
Comment (wachterjohannes) [reply bin/phpunit:8]: 🎉

File: config/packages/prod/monolog.yaml (hunk 3 above)
Comment (wachterjohannes) [reply config/packages/prod/monolog.yaml:11]: 👍

**Review by wachterjohannes — APPROVED** [review]

**Pull request merged**